### PR TITLE
Clear 'SecretKeySet' on drop.

### PR DESCRIPTION
**Pull request by DrPeterVanNostrand**
- Added `impl Drop for Poly` which fills the `coeff` vector with `Fr::zero()`.
- Added `impl Drop for BivarPoly` which fills the `coeff` vector with `Fr::zero()` and resets the `degree`.

Closes Issue #139

### CODE REVIEW

**Comment by afck** [comment]
Sorry about that! That should have been an `ln -sf` [in the Travis file](https://github.com/poanetwork/hbbft/pull/138/files#diff-354f30a63fb0907d4ad57269548329e3R17).

**Comment by DrPeterVanNostrand** [comment]
Np, I'll add `Drop` to `BivarPoly`. Should I change the travis file with this PR?

**Comment by afck** [comment]
Sure, we can make the same change in every PR. Whatever gets merged first fixes it.

File: src/crypto/poly.rs [hunk 1]
```diff
@@ -327,6 +340,20 @@ pub struct BivarPoly {
     coeff: Vec<Fr>,
 }
 
+impl Drop for BivarPoly {
+    fn drop(&mut self) {
+        let degree_ptr = self.degree as *mut usize;
+        let coeff_ptr = self.coeff.as_mut_ptr();
+        unsafe {
+            write_volatile(degree_ptr, 0);
```
Comment (afck) [inline src/crypto/poly.rs]: That's unnecessary; the degree is in the commitment, too, so it can't be a secret anyway.

**Pull request merged**